modules/userService.js

```diff
@@ -1,6 +1,9 @@
 var mongoose = require("mongoose");
 var bcrypt = require("bcryptjs");
 
+//EnsureIndex deprecation replaced by with useCreateIndex
+mongoose.set('useCreateIndex',true);
+
```

Comment:
This is good, but probably better to keep mongoose settings in the options parameter below in the `mongoose.connect()`